RootSeq.m

```diff
@@ -159,6 +159,48 @@ function defaultBranch(self, target)
             % with the sequence as the argument.
             self.after_branch_cbs{end + 1} = cb;
         end
+        %% Methods that call topLevel methods to allow for more convenient and linear API usage
+        function bseq = newBasicSeq(self, cb, varargin)
```

Comment:
Unless we are going to create thousands of basic sequence the performance here is not that important. You can either do

```matlab
function bseq = newBasicSeq(self, varargin)
    bseq = newBasicSeq(self.topLevel, varargin{:});
end
```

here or remove the implementation in `ExpSeq`.

Same with most of the other functions.